apps/system/js/notifications.js

```diff
@@ -456,10 +468,31 @@ var NotificationScreen = {
     message.classList.add('detail');
     var messageContent = document.createElement('div');
     messageContent.classList.add('detail-content');
-    messageContent.textContent = detail.text;
+    // As per the UX,in text messages notification text will be name of
+    // the sender and the text of message.
+    if(detail.manifestURL === smsManifestUrl) {
+      messageContent.textContent = detail.title + ': ' + detail.text;
+    } else {
+      messageContent.textContent = detail.text;
+    }
     messageContent.setAttribute('dir', 'auto');
     message.appendChild(messageContent);
     notificationNode.appendChild(message);
+    } else {
+      // in case if notification is already present on the screen, its counter
+      // value will be incremented.
+      var notificationNode = document.querySelector('.notification');
+      var countVal = this.unreadNotifications[cIndex].count;
+      this.unreadNotifications[cIndex].count = ++countVal;
+      document.getElementById(manifestURL).querySelector('.detail-count').textContent = countVal;
+      var message = document.getElementById(manifestURL).querySelector('.detail-content').textContent;
+      if(detail.manifestURL === smsManifestUrl) {
+        message = message + "; " + detail.title + ': ' + detail.text;
```

Comment:
I'm not sure about requirements for lockScreen notifications, but for homescreen there is a requirement to decrement counter if user reads for example one of sms messages, so in case of concatenated text from several messages you will need to extract parts of string which belongs to already read message from this common string. This is one more reason to store required notifications data not only in html nodes, but, probably in additional array of objects.